internal/dao/rbac_policy.go

```diff
@@ -159,18 +160,30 @@ func (p *Policy) fetchRoleBindingSubjects(kind, name string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	ns, n := client.Namespaced(name)
 	ss := make([]string, 0, len(rbs))
 	for _, rb := range rbs {
 		for _, s := range rb.Subjects {
-			if s.Kind == kind && s.Name == name {
-				ss = append(ss, rb.RoleRef.Kind+":"+rb.Name)
+			if isSameSubject(kind, ns, n, s) {
+				ss = append(ss, rb.RoleRef.Kind+":"+rb.RoleRef.Name)
```

Comment:
The change from `rb.Name` to `rb.RoleRef.Name` fixes the role/rolebinding naming issue.